app/src/main/java/com/censocustody/android/data/EncryptionManager.kt

```
@@ -305,16 +309,13 @@ class EncryptionManagerImpl @Inject constructor(
         }.flatten()
     }
 
-    fun handleRecoverRootSeed(
+    override fun recoverRootSeedFromShards(
         shards: List<Shard>,
         ancestors: List<AncestorShard>,
-        privateKeys: List<String>
+        email: String
     ): ByteArray {
+        val privateKeyMap = getMapOfDeviceKeys(email = email)
```

not on this line but on line 331 - I know you copied this code from the kotlin mock code in prime, but we might want some additional checks around indexing into arrays or the !! there.  maybe that should be `firstOrNull().toByteArrayNoSign(64)` and if rootseed ends being null maybe throw an exception/send a ray gun alert.